### PR TITLE
Add a style guide

### PR DESCRIPTION
**Pull request by zanieb**

### CODE REVIEW

File: STYLE.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Style guide
+
+_The following is a work-in-progress style guide for our user-facing messaging in the CLI output and documentation_.
+
+## General
+
+1. Use of "e.g." and "i.e." should always be wrapped in commas, e.g., as shown here.
+1. Em-dashes are okay, but not recommended when using monospace fonts. Use "—", not "--" or "-".
```
Comment (charliermarsh) [inline STYLE.md:8]: I think you tend to put spaces around em dashes.

File: STYLE.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+### Code blocks
+
+1. All code blocks should have a language marker.
+1. When using `console` syntax, use `$` to indicate commands — everything else is output.
+1. Do not use the `bash` syntax when displaying command output.
+1. Command output should rarely be included — it's hard to keep up to date.
+
+## CLI
+
+1. Do not use periods at the end of sentences :)
```
Comment (charliermarsh) [inline STYLE.md:57]: Unless the message spans more than a single sentence.

File: STYLE.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+1. Command output should rarely be included — it's hard to keep up to date.
+
+## CLI
+
+1. Do not use periods at the end of sentences :), unless the message spans more than a single sentence.
+1. May use the second-person point of view, e.g., "Did you mean...?".
+
+### Colors and style
+
+1. All CLI output should be interpretable and understandable _without_ the use of color and other styling. (For example: even if a command is rendered in green, wrap it in backticks.)
```
Comment (konstin) [inline STYLE.md:79]: I'd say must, piping to a text file should not lose any information.

File: STYLE.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+1. Use less than and greater than symbols to wrap bare URLs, e.g., <https://astral.sh> (unless it is an example; then, use backticks).
+1. Avoid bare URLs outside of reference documentation, prefer labels, e.g., `[name](url)`.
+1. If a message ends with a single relevant value, precede it with a colon, e.g., `This is the value: value`. If the value is a literal, wrap it in backticks.
+
+## Styling uv
+
+Just uv, please.
+
+1. Do not escape with backticks, e.g., `uv`, unless referring specifically to the `uv` executable.
+1. Do not capitalize, e.g., "Uv".
```
[the remark this answers is not in the document]
Comment (zanieb) [reply STYLE.md:21]:
```suggestion
1. Do not capitalize, e.g., "Uv", even at the beginning of a sentence.
```

File: STYLE.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+1. Do not use periods at the end of sentences :), unless the message spans more than a single sentence.
+1. May use the second-person point of view, e.g., "Did you mean...?".
+
+### Colors and style
+
+1. All CLI output should be interpretable and understandable _without_ the use of color and other styling. (For example: even if a command is rendered in green, wrap it in backticks.)
+1. In general, we use: green for success, red for error, yellow for warning, and cyan for hints; cyan for file paths, cyan for salient user-facing values (e.g., a package name in a message), and green for commands. 
+1. `NO_COLOR` must be respected when using any colors or styling.
+1. `UV_NO_PROGRESS` must be respected when using progress-styling like bars or spinners.
```
[the remark this answers is not in the document]
Comment (zanieb) [reply STYLE.md:81]: Idk it seems worth codifying while we're talking about style? Sometimes you are not using our standard printer styling e.g. in the `help` pager.

File: STYLE.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+### Colors and style
+
+1. When instructing the user to execute a command, it should be bolded. 
+1. `NO_COLOR` must be respected when using any colors or styling.
+1. `UV_NO_PROGRESS` must be respected when using progress-styling like bars or spinners.
+
+### Logging
+
+1. `warn`, `info`, `debug`, and `trace` logs are all shown with the `--verbose` flag.
```
[the remark this answers is not in the document]
Comment (zanieb) [reply STYLE.md:93]: Not unless that changed recently? It's dependent entirely on the `RUST_LOG` level right now. https://github.com/astral-sh/uv/issues/1569

Great prompts for the rest!

File: STYLE.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+1. Should assume no previous knowledge.
+1. Should refer to relevant concept documentation.
+1. Should have a clear flow.
+1. Should be followed by a clear call to action.
+1. Should cover the basic behavior needed to get started.
+1. Should not cover behavior in detail.
+1. Should not enumerate all possibilities.
+1. Should avoid linking to reference documentation unless not covered in a concept document.
+1. May generally ignore platform-specific behavior
+1. May be written from second-person point of view, though we prefer third-person at this time.
```
[the remark this answers is not in the document]
Comment (zanieb) [reply STYLE.md:48]: Yes I agree. I'm on the fence here. Basically I think it's okay for guides to be second-person but so far I've written everything in third-person.

File: STYLE.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
+# Style guide
+
+_The following is a work-in-progress style guide for our user-facing messaging in the CLI output and documentation_.
+
+## General
+
+1. Use of "e.g." and "i.e." should always be wrapped in commas, e.g., as shown here.
+1. Em-dashes are okay, but not recommended when using monospace fonts. Use "—", not "--" or "-".
```
[the remark this answers is not in the document]
Comment (zanieb) [reply STYLE.md:8]: What's an input file?

File: STYLE.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+### Colors and style
+
+1. When instructing the user to execute a command, it should be bolded. 
+1. `NO_COLOR` must be respected when using any colors or styling.
+1. `UV_NO_PROGRESS` must be respected when using progress-styling like bars or spinners.
+
+### Logging
+
+1. `warn`, `info`, `debug`, and `trace` logs are all shown with the `--verbose` flag.
+1. `warn_user` and `warn_user_once` are shown without the `--verbose `flag.
```
[the remark this answers is not in the document]
Comment (charliermarsh) [reply STYLE.md:94]: Not sure I can answer that cleanly, but one useful criterion: is the message user-actionable?

File: STYLE.md (hunk 3 above)
Comment (charliermarsh) [reply STYLE.md:79]: Yes, sorry, must.

File: STYLE.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (zanieb) [reply STYLE.md:94]: Yeah no clean answer, that's a great guideline — it's about severity and actionability imo.

File: STYLE.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (konstin) [reply STYLE.md:8]: Our plain text files, e.g. markdown files, as opposed to rich output files such as html. For example, latex has `\textemdash` and converts that to a real em dash in the pdf output.

File: STYLE.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (konstin) [reply STYLE.md:81]: Could we have an abstraction that codifies those rules in one place?

File: STYLE.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (konstin) [reply STYLE.md:93]: On any level, i'm only seeing trace logs when `RUST_LOG=trace` is set. We probably should link to #1569 saying that the above is how it currently is but this should change with #1569

[the remark this answers is not in the document]
Comment (zanieb) [reply STYLE.md:99]: I added a note. Hesitant to link to the issue, but now the issue is linked to here and we should update the style guide when we change `-vv` behavior.

File: STYLE.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (zanieb) [reply STYLE.md:8]: In documentation? I have a pretty strong preference for being able to use em dashes in the markdown.

File: STYLE.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (zanieb) [reply STYLE.md:87]: I mean yeah but you still need to use the abstraction. Why the opposition here? I don't think these are harmful to note.

[the remark this answers is not in the document]
Comment (konstin) [reply STYLE.md:87]: I'm not against having them in the style guide as long as long as we generally have abstractions for this

File: STYLE.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (konstin) [reply STYLE.md:8]: Non-unicode characters are a nuisance to enter, esp. across keyboard layouts, and when they look like an ascii characters they tend to cause unnecessary debugging

File: STYLE.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+1. Avoid bare URLs outside of reference documentation, prefer labels, e.g., `[name](url)`.
+1. If a message ends with a single relevant value, precede it with a colon, e.g., `This is the value: value`. If the value is a literal, wrap it in backticks.
+
+## Styling uv
+
+Just uv, please.
+
+1. Do not escape with backticks, e.g., `uv`, unless referring specifically to the `uv` executable.
+1. Do not capitalize, e.g., "Uv", even at the beginning of a sentence.
+1. Do not uppercase, e.g., "UV".
```
Comment (samypr100) [inline STYLE.md:22]: I assume the exception to this is with env vars?

Comment (zanieb) [reply STYLE.md:22]: Thanks, that seems like a good thing to note .
```suggestion
1. Do not uppercase, e.g., "UV", unless referring to an environment variable, e.g., `UV_PYTHON`.
```

**Review by charliermarsh — APPROVED** [review]
I'm fine to merge this whenever and iterate.

**Comment by zanieb** [comment]
Let's send it — lots more to do here though.

**Pull request merged**